static/js/subs.js

```diff
@@ -222,15 +222,14 @@ exports.update_settings_for_subscribed = function (sub) {
     var settings_button = settings_button_for_sub(sub).removeClass("unsubscribed").show();
     $('.add_subscribers_container').show();
 
+    exports.add_sub_to_table(sub);
     if (button.length !== 0) {
         exports.rerender_subscribers_count(sub, true);
 
         button.toggleClass("checked");
         settings_button.text(i18n.t("Unsubscribe"));
 
         stream_edit.add_me_to_member_list(sub);
-    } else {
```

Comment:
This else shouldn't be removed, this checks if subscription template is already rendered or not. Same way `add_sub_to_table` is also shouldn't called every time `update_settings_for_subscribed` gets called.